### PR TITLE
Add krb5 patches

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: krb5/Patch-for-Kerberos-5-1.16.1.patch [hunk 1]
```diff
@@ -0,0 +1,1949 @@
+From 6a11edab7fa3813ee62179848672abed5ac79ec3 Mon Sep 17 00:00:00 2001
+From: Juliusz Sosinowicz <juliusz@wolfssl.com>
+Date: Tue, 14 Sep 2021 13:37:02 +0200
+Subject: [PATCH] Patch for Kerberos 5 1.16.1
+
+- Use `PKG_CHECK_MODULES` and `PKG_CHECK_VAR` to find wolfSSL and configure compilation options
+- Rename `WANT_READ` and `WANT_WRITE` to `K5_WANT_READ` and `K5_WANT_WRITE` respectively to avoid conflicts with wolfSSL
+- Remove Camellia tests as wolfSSL does not support it in the EVP layer
```
Comment (haydenroche5) [inline krb5/Patch-for-Kerberos-5-1.16.1.patch:8]: It looks like it's still trying to build some of the Camellia code (and failing) on my system:

```
make[4]: Entering directory '/home/hayden/tmp/krb5-1.16.1/src/lib/crypto/openssl/enc_provider'
gcc -fPIC -DSHARED -DHAVE_CONFIG_H  -I../../../../include -I../../../../include -I./../../krb -I./.. -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE  -g -O2 -I/usr/local/include -I/usr/local/include/wolfssl -I/usr/local/include -I/usr/local/include/wolfssl -Wall -Wcast-align -Wshadow -Wmissing-prototypes -Wno-format-zero-length -Woverflow -Wstrict-overflow -Wmissing-format-attribute -Wmissing-prototypes -Wreturn-type -Wmissing-braces -Wparentheses -Wswitch -Wunused-function -Wunused-label -Wunused-variable -Wunused-value -Wunknown-pragmas -Wsign-compare -Werror=uninitialized -Wno-maybe-uninitialized -Werror=pointer-arith -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=discarded-qualifiers -Werror=implicit-int -Werror=declaration-after-statement -Werror-implicit-function-declaration -pthread  -c des.c -o des.so.o && mv -f des.so.o des.so
gcc -fPIC -DSHARED -DHAVE_CONFIG_H  -I../../../../include -I../../../../include -I./../../krb -I./.. -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE  -g -O2 -I/usr/local/include -I/usr/local/include/wolfssl -I/usr/local/include -I/usr/local/include/wolfssl -Wall -Wcast-align -Wshadow -Wmissing-prototypes -Wno-format-zero-length -Woverflow -Wstrict-overflow -Wmissing-format-attribute -Wmissing-prototypes -Wreturn-type -Wmissing-braces -Wparentheses -Wswitch -Wunused-function -Wunused-label -Wunused-variable -Wunused-value -Wunknown-pragmas -Wsign-compare -Werror=uninitialized -Wno-maybe-uninitialized -Werror=pointer-arith -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=discarded-qualifiers -Werror=implicit-int -Werror=declaration-after-statement -Werror-implicit-function-declaration -pthread  -c des3.c -o des3.so.o && mv -f des3.so.o des3.so
gcc -fPIC -DSHARED -DHAVE_CONFIG_H  -I../../../../include -I../../../../include -I./../../krb -I./.. -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE  -g -O2 -I/usr/local/include -I/usr/local/include/wolfssl -I/usr/local/include -I/usr/local/include/wolfssl -Wall -Wcast-align -Wshadow -Wmissing-prototypes -Wno-format-zero-length -Woverflow -Wstrict-overflow -Wmissing-format-attribute -Wmissing-prototypes -Wreturn-type -Wmissing-braces -Wparentheses -Wswitch -Wunused-function -Wunused-label -Wunused-variable -Wunused-value -Wunknown-pragmas -Wsign-compare -Werror=uninitialized -Wno-maybe-uninitialized -Werror=pointer-arith -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=discarded-qualifiers -Werror=implicit-int -Werror=declaration-after-statement -Werror-implicit-function-declaration -pthread  -c rc4.c -o rc4.so.o && mv -f rc4.so.o rc4.so
: updated OBJS.SH
gcc -fPIC -DSHARED -DHAVE_CONFIG_H  -I../../../../include -I../../../../include -I./../../krb -I./.. -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE  -g -O2 -I/usr/local/include -I/usr/local/include/wolfssl -I/usr/local/include -I/usr/local/include/wolfssl -Wall -Wcast-align -Wshadow -Wmissing-prototypes -Wno-format-zero-length -Woverflow -Wstrict-overflow -Wmissing-format-attribute -Wmissing-prototypes -Wreturn-type -Wmissing-braces -Wparentheses -Wswitch -Wunused-function -Wunused-label -Wunused-variable -Wunused-value -Wunknown-pragmas -Wsign-compare -Werror=uninitialized -Wno-maybe-uninitialized -Werror=pointer-arith -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=discarded-qualifiers -Werror=implicit-int -Werror=declaration-after-statement -Werror-implicit-function-declaration -pthread  -c aes.c -o aes.so.o && mv -f aes.so.o aes.so
gcc -fPIC -DSHARED -DHAVE_CONFIG_H  -I../../../../include -I../../../../include -I./../../krb -I./.. -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE  -g -O2 -I/usr/local/include -I/usr/local/include/wolfssl -I/usr/local/include -I/usr/local/include/wolfssl -Wall -Wcast-align -Wshadow -Wmissing-prototypes -Wno-format-zero-length -Woverflow -Wstrict-overflow -Wmissing-format-attribute -Wmissing-prototypes -Wreturn-type -Wmissing-braces -Wparentheses -Wswitch -Wunused-function -Wunused-label -Wunused-variable -Wunused-value -Wunknown-pragmas -Wsign-compare -Werror=uninitialized -Wno-maybe-uninitialized -Werror=pointer-arith -Werror=int-conversion -Werror=incompatible-pointer-types -Werror=discarded-qualifiers -Werror=implicit-int -Werror=declaration-after-statement -Werror-implicit-function-declaration -pthread  -c camellia.c -o camellia.so.o && mv -f camellia.so.o camellia.so
camellia.c:29:10: fatal error: openssl/camellia.h: No such file or directory
   29 | #include <openssl/camellia.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

**Comment by julek-wolfssl** [comment]
This PR now depends on https://github.com/wolfSSL/wolfssl/pull/4670.

File: krb5/Patch-for-Kerberos-5-1.16.1.patch [hunk 2]
```diff
@@ -0,0 +1,1949 @@
...
+
+Compile wolfSSL with:
+```
+./autogen.sh
+./configure --enable-krb
+make
+make install
+```
+
+Compile krb5 with
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply krb5/Patch-for-Kerberos-5-1.16.1.patch:24]: URL added.

File: krb5/Patch-for-Kerberos-5-1.16.1.patch (hunk 1 above)
Comment (julek-wolfssl) [reply krb5/Patch-for-Kerberos-5-1.16.1.patch:8]: I got a whole bunch of errors now. I'm not sure why (probably changes in wolfSSL master). I fixed the missing API/files in wolfSSL and updated the patch. It compiles for me and passes the tests.

File: krb5/Patch-for-Kerberos-5-1.19.2.patch [hunk 3]
```diff
@@ -0,0 +1,1945 @@
...
+
+Compile wolfSSL with:
+```
+./autogen.sh
+./configure --enable-krb
+make
+make install
+```
+
+Kerberos 1.16.1 maybe downloaded and extracted with:
```
Comment (haydenroche5) [inline krb5/Patch-for-Kerberos-5-1.19.2.patch:18]: maybe -> may be

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

**Comment by julek-wolfssl** [comment]
I've still got a bunch of test error that are hard to debug. Now I tried recreating them and I'm getting build errors? Its not an easy port.

**Comment by julek-wolfssl** [comment]
1.16.1 should pass tests. I'm not able to get everything passing with 1.19.2 (even when compiling against OpenSSL).

**Review by haydenroche5 — APPROVED** [review]

**Pull request merged**